src/hotspot/share/compiler/compilerDefinitions.cpp

```diff
@@ -497,6 +497,11 @@ bool CompilerConfig::check_args_consistency(bool status) {
                 "Invalid NonNMethodCodeHeapSize=%dK. Must be at least %uK.\n", NonNMethodCodeHeapSize/K,
                 min_code_cache_size/K);
     status = false;
+  } else if (InlineCacheBufferSize >= NonNMethodCodeHeapSize) {
+    jio_fprintf(defaultStream::error_stream(),
+                "Invalid InlineCacheBufferSize=" SIZE_FORMAT "K. Must be less than NonNMethodCodeHeapSize=" SIZE_FORMAT "K.\n",
+                InlineCacheBufferSize/K, NonNMethodCodeHeapSize/K);
```

Comment:
You need to check for alignment of the value. In [StubQueue()](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/code/stubs.cpp#L70) it is aligned up by `2*BytesPerWord` so the final value could be > `InlineCacheBufferSize`.

Reply:
I think the align up to 2*BytesPerWord is not really need, because BufferBlob::create already does its own alignment which will  make the final value > InlineCacheBufferSize.  BufferBlob::create uses the size as a minimum, not a maximum.
I don't think the above check should need to know the details of BufferBlob::create and StubQueue() alignment adjustments.  Having InlineCacheBufferSize near to NonNMethodCodeHeapSize is going the make the JVM fail in startup for other reasons, isn't it?  Maybe the max for InlineCacheBufferSize should be NonNMethodCodeHeapSize/2?

Reply:
Thank you for answering my questions, Dean.
Yes, non-nmethod section contains template Interpreter code, stubs and adapters. So it will fail immediately if no space left for it.  NonNMethodCodeHeapSize/2 is reasonable limit.

Reply:
Thanks @dean-long and @vnkozlov, I updated InlineCacheBufferSize limit to NonNMethodCodeHeapSize/2.